### PR TITLE
Do not use entity setter on services

### PR DESCRIPTION
**Pull request by mmadariaga**
Replace entity direct setters on services by dtos

#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [ ] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX)

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php [hunk 1]
```diff
@@ -76,6 +77,19 @@ public function execute(CompanyInterface $company)
         $domain = $this->entityTools
             ->persistDto($domainDto, $domain);
 
-        $company->setDomain($domain);
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php]: This a PRE_PERSIST event, I thought it was correct to use setters in LifeCycled entity here.

File: library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php [hunk 2]
```diff
@@ -76,6 +77,19 @@ public function execute(CompanyInterface $company)
...
+
+        $companyDto
+            ->setDomain($domainDto);
+
+        $this
+            ->entityTools
+            ->updateEntityByDto(
+                $company,
+                $companyDto
+            );
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php]: The previous service (library/Ivoz/Provider/Domain/Service/Domain/UpdateByBrand.php ) uses persistDto while this uses updateEntityByDto. Is POST/PRE persist the reason behind this?

File: library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php (hunk 1 above)
Comment (mmadariaga) [reply library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php]: Entity setters will be protected functions soon, that's why I'm replacing them here

File: library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php (hunk 2 above)
Comment (mmadariaga) [reply library/Ivoz/Provider/Domain/Service/Domain/UpdateByCompany.php]: yes it is

**Comment by ironArt3mis** [comment]
Rebase completed.

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**